engine/access/ingestion/engine.go

```diff
@@ -337,9 +370,125 @@ func (e *Engine) processAvailableExecutionReceipts(ctx context.Context) error {
 		if err := e.handleExecutionReceipt(msg.OriginID, receipt); err != nil {
 			return err
 		}
+
+		// notify, to fetch and store transaction result error messages for block
+		e.txResultErrorMessagesChan <- receipt.BlockID
+	}
+}
+
+// processTransactionResultErrorMessages handles error messages related to transaction
+// results by reading from the error messages channel and processing them accordingly.
+//
+// This function listens for messages on the txResultErrorMessagesChan channel and
+// processes each transaction result error message as it arrives.
+//
+// No errors are expected during normal operation.
+func (e *Engine) processTransactionResultErrorMessages(ctx irrecoverable.SignalerContext, ready component.ReadyFunc) {
+	ready()
+
+	for {
+		select {
+		case <-ctx.Done():
+			return
+		case blockID := <-e.txResultErrorMessagesChan:
+			err := e.handleTransactionResultErrorMessages(ctx, blockID)
+			if err != nil {
+				// if an error reaches this point, it is unexpected
+				ctx.Throw(err)
+				return
+			}
+		}
 	}
 }
 
+// handleTransactionResultErrorMessages processes transaction result error messages for a given block ID.
+// It retrieves error messages from the backend if they do not already exist in storage.
+//
+// The function first checks if error messages for the given block ID are already present in storage.
+// If they are not, it fetches the messages from execution nodes and stores them.
+//
+// Parameters:
+// - ctx: The context for managing cancellation and deadlines during the operation.
+// - blockID: The identifier of the block for which transaction result error messages need to be processed.
+//
+// No errors are expected during normal operation.
+func (e *Engine) handleTransactionResultErrorMessages(ctx context.Context, blockID flow.Identifier) error {
+	if e.transactionResultErrorMessages != nil {
```

Comment:
> I do not think, that we need to check the storage for nil, as it should be set. Let it panic in case of a mistake.

The `nil` check for `transactionResultErrorMessages` is actually required because I’ve added a new CLI flag (`--store-tx-result-error-messages`) that, when set to false, prevents these messages from being stored in the protocol database. In such cases, `transactionResultErrorMessages` would be `nil`.